src/libstore/remote-store.cc

```diff
@@ -992,6 +955,82 @@ std::exception_ptr RemoteStore::Connection::processStderr(Sink * sink, Source *
     return nullptr;
 }
 
+
+struct FramedSink : nix::BufferedSink
```

Comment:
I would probably put this in `serialise.hh` too, or maybe put both the source in sink in a `serialise-framed.{cc,hh}`?

Reply:
If you do move out, every `conn->to` can be replaced with acessing an arbitrary `BufferedSink &` I think.

Reply:
Well, unlike `FramedSource` it depends on `RemoteStore` internals, so I figured it's appropriate to keep it here. It seems like it can be refactor to depend on `Sink` instead of `Connection`.

Reply:
Yeah I recommended `BufferedSink` so we can keep the `.flush()`. Think that makes the refactor "lossless" enough?